### DISCUSSION TITLE
Question: Is it ok for one slice to import another slice?

### DISCUSSION THREAD

**Discussion by ChristophP** [reduxjs/redux-toolkit#4727] [General]
I am having circular dependency problems in my root reducer which TypeScript complains about. It reads like

```
Circular definition of import alias `sliceB`
```

from a very simplified version of this code.

```ts
import sliceA from './slice-a';
import sliceB from './slice-b'; // this line gives a TS error
import { combineReducers } from '@reduxjs/toolkit';

const rootReducer = combineReducers({ sliceA, sliceB });
```

My root suspect is that in the codebase some slices import others in order to build selectors that aggregate data from different slices.

## So my question is ...

... is it even ok to use selectors of one slice in another slice? If not, what is the best way to structure selectors that compose values from multiple slices?

### Things I already tried

- Googling `Circular definition of import alias` but I couldn't find any description of what exactly causes this error
- Narrowing it down more but the codebase is quite large and I wasn't able to narrow it down a significant amount of effort

**Comment by markerikson** [comment]
Yes, it should be legal at both the types level and runtime level to do circular imports between slices.

It's certainly _possible_ to end up with either type or runtime errors depending on how the codebase is organized. I don't have concrete examples right this second.

**Reply by ChristophP** [reply]
Ok thx, I've had some slices import others but after a while got TS errors like "cannot resolve circular import alias".
It was difficult to track down what exactly caused it.
After a while I made sure no slices import others directly (which was mostly necessary for selectors) by creating a combined-selectors.ts file and made sure only that uses multiple slices.
That fixed my problems. Thx for the reply
